app/grandchallenge/algorithms/templates/algorithms/algorithmmodel_form.html

```diff
@@ -3,7 +3,7 @@
 {% load url %}
 
 {% block title %}
-    {% if object %}Update{% else %}Create{% endif %} An Algorithm Model - {{ block.super }}
+    Create Model - {{ algorithm.title }} - {{ block.super }}
 {% endblock %}

```

Comment:
Same as above: this template is used both on creation and on updating, right? Why did you remove that from the title?

Reply:
Check my comment above. It doesn't feel well to do it like this.